eos-fix-home-dir-permissions.service

```diff
@@ -0,0 +1,28 @@
+# Clean up any home directories which were created with
+# default permissions 755 rather than 700
+
+[Unit]
+Description=Fix existing home directory permissions from 755 to 700
+DefaultDependencies=no
+Conflicts=shutdown.target
+Wants=local-fs.target
+After=local-fs.target
+
+# Only run on updates
+Before=multi-user.target systemd-update-done.service
+ConditionNeedsUpdate=/etc
```

Comment:
Since this script is so simple and you're using a stamp file, I think I'd drop this and make the service much simpler. Once you drop `ConditionNeedsUpdate`, then you can have a normal service that doesn't need to run before `basic.target`. That simplifies things considerably since all the dependencies you care about will be automatically handled by systemd. I think it would reduce to (without comments):
```
[Unit]
Description=Fix existing home directory permissions from 755 to 700
ConditionPathExists=!/var/lib/eos-dir-mode-700

[Service]
Type=oneshot
RemainAfterExit=yes
ExecStart=/usr/sbin/eos-fix-home-dir-permissions

[Install]
WantedBy=multi-user.target
```